src/arrayMethodJoin.js

```diff
@@ -4,9 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const joinSeparator = separator;
+
+    const arr = [...this];
+
+    let res = '';
+
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] === null || arr[i] === undefined) {
+        arr[i] = '';
+      }
```

Comment:
You should do action if value not equal null or undefined
```suggestion
      if (arr[i] !== null || arr[i] !== undefined) {
        res += arr[i];
      }
```